app/copy_tree.py

```diff
@@ -1,34 +1,94 @@
+import grp
 import os
-import stat
+import pwd
 import shutil
+from dataclasses import dataclass
 from pathlib import Path
+from typing import Dict, Tuple
 
-import log_util
-from configs import config as cfg
-from ota_error import OtaErrorUnrecoverable
+from app import log_util
+from app.configs import config as cfg
 
 logger = log_util.get_logger(
     __name__, cfg.LOG_LEVEL_TABLE.get(__name__, cfg.DEFAULT_LOG_LEVEL)
 )
 
 
+@dataclass
+class PwdEntry:
+    pw_name: str
+    pw_uid: int
+
+    def __init__(self, pw_line: str) -> None:
+        """
+        typical pw_line format:
+            <name>:<passwd>:<uid>:<gid>:<comment>:<home>:<shell>
+            root:x:0:0:root:/root:/bin/bash
+        """
+        _parsed = pw_line.split(":")
+        self.pw_name = _parsed[0]
+        self.pw_uid = int(_parsed[2])
+
+
+@dataclass
+class GrpEntry:
+    gr_name: str
+    gr_id: int
+
+    def __init__(self, grp_line: str) -> None:
+        """
+        typical grp_line format:
+            <group_nmae>:<passwd>:<gid>:<list_of_username>
+            adm:x:4:syslog,autoware
+        """
+        _parsed = grp_line.split(":")
+        self.gr_name = _parsed[0]
+        self.gr_id = int(_parsed[2])
+
+
+def _load_dst_pwd(dst_pwd: Path) -> Dict[str, PwdEntry]:
+    # name -> uid
+    _uname_uid_mapping: Dict[str, PwdEntry] = {}
+    with open(dst_pwd, "r") as f:
+        for l in f:
+            _parsed = PwdEntry(l)
+            _uname_uid_mapping[_parsed.pw_name] = _parsed
+
+    return _uname_uid_mapping
+
+
+def _load_dst_grp(dst_group: Path) -> Dict[str, GrpEntry]:
+    # name -> uid
+    _gname_gid_mapping: Dict[str, GrpEntry] = {}
+    with open(dst_group, "r") as f:
+        for l in f:
+            _parsed = GrpEntry(l)
+            _gname_gid_mapping[_parsed.gr_name] = _parsed
+
+    return _gname_gid_mapping
+
+
 class CopyTree:
     def __init__(
         self,
-        src_passwd_file: Path,
-        src_group_file: Path,
```

Comment:
Can we support following case?
currnet: user name="ggc_user" UID=111, GID=222
OTA image: user name="ggc_user" UID=112, GID=223
persistent file owned by ggc_user should be copied to standby bank with UID=112 and GID=223.

Reply:
Is this function refactoring required for this PR??

Reply:
I will revert the change in the copy_tree.

Reply:
Yes, please. If you have any other uncecesary changes **FOR THIS PR**, they should be also reverted.

Reply:
OK, but new implementation should be done on another PR, please.

Reply:
Since this PR is huge.

Reply:
revert to previous implementation at [19aa212](https://github.com/tier4/ota-client/pull/136/commits/19aa212a191b31024eb6e7752efe4f0211ee89a1)